documentation/docs/client/moderation_tools.md

```diff
@@ -0,0 +1,195 @@
+
+```kotlin
+client.flagMessage("message-id").enqueue { result -> 
+    if (result.isSuccess) { 
+        // Message was flagged 
+        val flag: Flag = result.data() 
+    } else { 
+        // Handle result.error() 
+    } 
+} 
+ 
+client.flagUser("user-id").enqueue { result -> 
+    if (result.isSuccess) { 
+        // User was flagged 
+        val flag: Flag = result.data() 
+    } else { 
+        // Handle result.error() 
+    } 
+}
+```
+
+## Muting a User
+
+Any user is allowed to mute another user. Mutes are stored at user level and returned with the rest of the user information when `connectUser` is called. A user will be be muted until the user is `unmuted` or the mute is expired.
+
+```kotlin 
+client.muteUser("user-id").enqueue { result -> 
+    if (result.isSuccess) { 
+        // User was muted 
+        val mute: Mute = result.data() 
+    } else { 
+        // Handle result.error() 
+    } 
+} 
+ 
+client.unmuteUser("user-id").enqueue { result -> 
+    if (result.isSuccess) { 
+        // User was unmuted 
+    } else { 
+        // Handle result.error() 
+    } 
+}
+```
+
+After muting a user messages will still be delivered via web-socket. Implementing business logic such as hiding messages from muted users or display them differently is left to the developer to implement.
+
+> Messages from muted users are not delivered via push (APN/Firebase)
+
+## Banning a User
+
+Users can be banned from an app entirely or from a channel. When a user is banned, it will not be allowed to post messages until the ban is removed or expired but it will be able to connect to Chat and to channels as before.
+
+In most cases, only admins or moderators are allowed to ban other users from a channel.
+
+| Name | Type | Description | Default | Optional |
+| :--- | :--- | :--- | :--- | :--- |
+| timeout | Int | The timeout in minutes until the ban is automatically expired. | 	no limit | &check; |
+| reason | String | The reason that the ban was created. | | &check; |
+
+> Banning a user from all channels can only be done using server-side auth.
+
+```kotlin
+// Ban user for 60 minutes from a channel 
+channelClient.banUser(targetId = "user-id", reason = "Bad words", timeout = 60).enqueue { result -> 
+    if (result.isSuccess) { 
+        // User was banned 
+    } else { 
+        // Handle result.error() 
+    } 
+} 
+ 
+channelClient.unBanUser(targetId = "user-id").enqueue { result -> 
+    if (result.isSuccess) { 
+        // User was unbanned 
+    } else { 
+        // Handle result.error() 
+    } 
+}
+```
+
+## Shadow Banning a User
+
+Users can be shadow banned from an app entirely or from a channel. When a user is shadow banned, it will still be allowed to post messages, but any message sent during, will have `shadowed: true` field. However, this will be invisible to the author of the message.
```

Comment:
```suggestion
Users can be shadow banned from an app entirely or just from a single channel. When a user is shadow banned, they will still be allowed to post messages, but any message sent during, will have `shadowed: true` field. However, this will be invisible to the author of the message.
```

Reply:
`However, this will be invisible to the author of the message` - I think we should somehow change this sentence. What will be invisible? 
Maybe we can remove it - it's explained in next sentence: ` It's up to the client-side implementation to hide or otherwise handle these messages appropriately`

Reply:
Not sure. Need to investigate that. When the current user is shadow banned, then either:
- `message.shadowed` is always `false` for the messages of the current user
or
- `message.shadowed` is `true`  for the messages of the current user, but this should not be reflected in the UI